mono/metadata/threadpool.c

```diff
@@ -692,12 +613,18 @@ ves_icall_System_Threading_ThreadPool_GetAvailableThreadsNative (gint32 *worker_
 	if (!worker_threads || !completion_port_threads)
 		return;
 
-	mono_lazy_initialize (&status, initialize);
+	if (!mono_lazy_initialize (&status, initialize) || !mono_refcount_tryinc (&threadpool)) {
```

Comment:
I am confident that the 2 variables do not exclude each other, as the `refcount` has a longer lifecycle than `status`. It might be unnecessary, but I am not 100% confident about it, so I would prefer to keep as is for now, and once this PR is merged, we can explore removing `refcount`.

The logic is now that we cleanup the threadpool once all the threads have stopped running, we shouldn't need the `refcount`, as the `threadpool` variable is going to be alive for at least as long as there are threadpool threads running. BUT I don't have full confidence in our runtime shutdown mechanism, so I prefer to keep it that way for now.